oss/ossSocket.hpp

```diff
@@ -77,7 +77,17 @@ class _ossSocket {
 
   // --- deconstructor --- 
   ~_ossSocket() {
-    close(); 
+    close();
+  }
+  
+  void operator delete(void* ptr, size_t) {
+      // release the allocated space pointed by the ptr 
+      std::cout << "delete ptr " << ptr << std::endl; 
+      free(ptr);
+      // if we only free the space the pointer points to ptr value will not reseted to the nullptr automatically
+      // we need reset the pointer to the nullptr by manual 
```

Comment:
here i tried to reset the pointer after free the space allocated to it by free method. 

however cus this delete method's definition decides that the poitner only value passing instead of address passing. 
so, even though I reset it to the nullptr when calling end,the context switching to previous main context, the value of the poitner come back to the last address value so I have to reset it again in the main context.